parlai/core/image_featurizers.py

```diff
@@ -0,0 +1,145 @@
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+import parlai.core.build_data as build_data
+
+import torch
+from torch.autograd import Variable
+import torchvision
+import torchvision.transforms as transforms
+import torch.nn as nn
+
+import os
+import copy
+import numpy as np
+from PIL import Image
+
+_greyscale = '  .,:;crsA23hHG#98&@'
+
+class ImageLoader():
+	"""Extract image feature using pretrained CNN network.
+	"""	
+	@staticmethod
+	def add_cmdline_args(argparser):
+		argparser.add_arg('--image-cnntype', type=str, default='resnet152',
```

Comment:
there are two parameters used for image feature extraction. --image_cnntype (which specify the CNN type), and --image_layernum (which specify which layer of feature we need). Usually, we use output of fully connected layer as global image feature, or output of last convolutional layer as spatial image feature or others. Previously, I'm thinking we can let user to set the --image_cnntype or --image_layernum, and --image-mode just specify which folder to store the feature.